unblob/processing.py

```diff
@@ -14,20 +14,21 @@ def process_file(
     root: Path,
     path: Path,
     extract_root: Path,
-    depth: int,
+    max_depth: int,
+    current_depth: int = 0,
 ):
-    is_initial_file = DEFAULT_DEPTH == depth
+    is_initial_file = current_depth == 0
     log = logger.bind(path=path)
     log.info("Start processing file", _absolute_path=is_initial_file)
 
-    if depth <= 0:
+    if current_depth == max_depth:
```

Comment:
It would be a pretty nasty bug if we would somehow increase it by 2! :smile: 

But I get the intention rewrote to this:

```python
    if current_depth >= max_depth:
        log.info("Reached maximum depth, stop further processing")
        return
```